### DISCUSSION TITLE
How to customize a patched font's family name?

### DISCUSSION THREAD

**Discussion by pa-0** [ryanoasis/nerd-fonts#1754] [Q&A] [answered]
Is that what the `--name` flag is for? Or is there no flag for accomplishing this? 

#### An Illustrative Example:

Patching, for instance, the Input Mono font, can I pass `--name "Input"` to the script so that the resulting fonts are output with names like **_`Input Nerd Font Mono`_**, **_`Input Nerd Font Propo`_**, etc.?

**Comment by Finii** [comment, marked as answer]
https://github.com/ryanoasis/nerd-fonts/wiki/ScriptOptions

Just try? Possibly together with `--dry --debug 2` to just see how the names would come out.

```console
$ fontforge font-patcher --dry --debug 2 --name TEST-Bold src/unpatched-fonts/Agave/Agave-Regular.ttf
Nerd Fonts Patcher v3.3.0-22 (4.16.1) (ff 20230101)
DEBUG: Naming mode 1
DEBUG: Monospace check: Panose says "monospaced"; glyph-width-mono True
DEBUG: Extended glyphs wider bounding box than basic glyphs
DEBUG: Final font cell dimensions 1024 w x 2048 h
Done with Patch Sets, generating font...
WARNING: Possible problem with the weight metadata detected, check with --debug
DEBUG: Weight approximations: OS2/PS/Name: 400/400/700 (from 400/'Book'/'Bold')
DEBUG: =====> Family (ID 1)      ok       ( 4 <=31): TEST
DEBUG: =====> SubFamily (ID 2)   ok       ( 4 <=31): Bold
DEBUG: =====> Fullname (ID 4)    ok       ( 9 <=63): TEST Bold
DEBUG: =====> PSN (ID 6)         ok       ( 9 <=63): TEST-Bold
DEBUG: =====> Filename 'TEST-Bold.ttf'
```

**Reply by pa-0** [reply]
Thank you! That demonstration was super helpful. I actually did try it (in a sense). I've been working on using the Delugia Code repository as a template to automate patching other fonts.  It was a bit of a leanring curve, but I've realized that (with the rename_font script set up there) the original repository does not follow the typical font-patching workflow, so my tests with the `--name` flag were not producing results as expected. 


I think I've worked out most of the kinks though, and now have a pretty easy way to keep fonts patched through github actions as new versions are released -- at least, fonts that are hosted in git repositories. To fully automate though, I need to account for certain fonts requiring extra flags (like `--has-no-italic`).

**Reply by Finii** [reply]
Ah, when Adam created Delugia the `font-patcher` did not yet have a `--name` flag; nowadays it would not be needed anymore.
Which reminds me that I believe I said over there that I would update Delugia to Nerd Font HEAD. :roll_eyes: 

The lines that call the renamer could be translated into direct `--name` calls like this:
(i.e. use the target filename and remote the suffix)

```console
./do_generate 14 --powerline --mono CascadiaMonoPL-BoldItalic.ttf         DelugiaMonoPL-BoldItalic.ttf   "Delugia PL Mono"
```

```console
$ fontforge font-patcher src/unpatched-fonts/Agave/Agave-Regular.ttf --dry --debug 2 --name DelugiaMonoPL-BoldItalic 2>/dev/null
Nerd Fonts Patcher v3.3.0-10 (4.16.1) (ff 20230101)
DEBUG: Naming mode 1
DEBUG: Monospace check: Panose says "monospaced"; glyph-width-mono True
DEBUG: Extended glyphs wider bounding box than basic glyphs
DEBUG: Final font cell dimensions 1024 w x 2048 h
Done with Patch Sets, generating font...
WARNING: Possible problem with the weight metadata detected, check with --debug
DEBUG: Weight approximations: OS2/PS/Name: 400/400/700 (from 400/'Book'/'Bold')
DEBUG: =====> Family (ID 1)      ok       (15 <=31): Delugia Mono PL
DEBUG: =====> SubFamily (ID 2)   ok       (11 <=31): Bold Italic
DEBUG: =====> Fullname (ID 4)    ok       (27 <=63): Delugia Mono PL Bold Italic
DEBUG: =====> PSN (ID 6)         ok       (24 <=63): DelugiaMonoPL-BoldItalic
DEBUG: =====> Filename 'DelugiaMonoPL-BoldItalic.ttf'
```

Name parts look good, iirc.

----

> I need to account for certain fonts requiring extra flags (like --has-no-italic).

We could think about something to put all the individual `config.cfg` into a "database" in the patcher, so that these are automatically be used, also in your use case :thinking:

**Reply by pa-0** [reply]
Yes I figured as much after more trial and error then I'd like to admit - [insert truism about hindsight here]. 😆  

The config file definiely sounds like a useful propositiion.  Thanks again for the assist.

It's in a very hacky state currently, but once I've smoothed out all the edges, I plan to create a more generic template based on this repository:

https://github.com/pa-0/ReaditMono-NF
